### PR TITLE
[terra-application-navigation]Callback added to notify users for Drawer Menu state change

### LINKED ISSUE #960: [terra-application-navigation] Add callback to provide the state of the Drawer menu to consumers
# Feature Request

## Description
There is a request from a consumer to know what state the navigation drawer menu is in. We should add a callback function that reports the state of the navigation drawer as it is either changed or reset.

The prop should be named:
OnDrawerMenuStateChange

### PR DESCRIPTION
**Pull request by lokesh-0813**
fixes #960

### CODE REVIEW

**Comment by mjhenkes** [comment]
Add a test please

File: packages/terra-application-navigation/src/ApplicationNavigation.jsx [hunk 1]
```diff
@@ -144,6 +149,9 @@ const ApplicationNavigation = ({
   function updateDrawerIsOpen(value) {
     drawerMenuIsOpenRef.current = value;
     setDrawerMenuIsOpen(value);
+    if (onDrawerMenuStateChange) {
+      onDrawerMenuStateChange(value);
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/ApplicationNavigation.jsx]: Can we add a unit test for this?

**Comment by lokesh-0813** [comment]
> Add a test please

Added here 1376560be921551f9527d18db6352063b720f59d

File: packages/terra-application-navigation/src/ApplicationNavigation.jsx (hunk 1 above)
Comment (lokesh-0813) [reply packages/terra-application-navigation/src/ApplicationNavigation.jsx]: Added here 1376560be921551f9527d18db6352063b720f59d

Comment (mjhenkes) [reply packages/terra-application-navigation/src/ApplicationNavigation.jsx]: A jest test may have been more appropriate, but this works too.

**Pull request merged**